File: back/src/main/java/org/example/control_user/service/UserService.java
package org.example.control_user.service;

import org.example.control_user.entity.User;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface UserService {
    User saveUser(User user);

    Optional<User> checkUser(String name, String pass);

    List<User> getUsers();

    Optional<User> getUser(Long id);

    void deleteUser(Long id);

    void updateUser(User user);
}

File: front/src/services/UserService.js
import axios from 'axios';

const USER_API_BASE_URL = "http://89.117.109.48:8080/v1/user/";

class UserService {

    getUsers() {
        return axios.get(USER_API_BASE_URL);
    }

    createUser(user) {
        return axios.post(USER_API_BASE_URL, user);
    }

    getUserByName(name, pass) {
        return axios.get(USER_API_BASE_URL + 'check?name=' + name + '&pass=' + pass);
    }

    getUserById(userId) {
        return axios.get(USER_API_BASE_URL + userId);
    }

    updateUser(user) {
        return axios.put(USER_API_BASE_URL, user);
    }

    deleteUser(userId) {
        return axios.delete(USER_API_BASE_URL + userId);
    }
}

export default new UserService()